File: home/dot_local/share/lazyvim/lua/lazyvim_capabilities/contract.lua
return function(definition)
	assert(type(definition) == "table" and type(definition.id) == "string", "capability requires a string id")
	definition.requires = definition.requires or {}
	definition.supports = definition.supports or function()
		return true
	end
	definition.enhancements = definition.enhancements or {}
	return definition
end

File: home/dot_local/share/lazyvim/capabilities/contract.mjs
export function defineCapability(definition) {
  if (!definition?.id || typeof definition.id !== "string") {
    throw new Error("A capability must have a string id");
  }
  return Object.freeze({
    requires: [],
    supports: () => true,
    enhancements: {},
    ...definition,
  });
}
